### PR TITLE
Modifying bonus block custom object from editor

### PR DESCRIPTION
**Pull request by Vankata453**
Adds the ability to edit a bonus block custom object directly from the editor. Previously, this had to be done via editing the file.

Additionally fixes a segmentation fault when dropping more than 1 custom object.

### CODE REVIEW

**Review by mrkubax10 — APPROVED** [review]

File: src/editor/object_option.cpp [hunk 1]
```diff
@@ -513,8 +524,9 @@ ObjectSelectObjectOption::to_string() const
 void
 ObjectSelectObjectOption::add_to_menu(Menu& menu) const
 {
-  menu.add_entry(get_text(), [pointer = m_value_pointer, parent = m_parent]() {
-    MenuManager::instance().push_menu(std::make_unique<ObjectSelectMenu>(*pointer, parent));
+  menu.add_entry(get_text(), [pointer = m_value_pointer, get_objects_param = m_get_objects_param,
```
Comment (MatusGuy) [inline src/editor/object_option.cpp:527]: Wh... Heh... Ha... Wha???

Comment (mrkubax10) [reply src/editor/object_option.cpp:527]: What do you mean?

**Review by MatusGuy — APPROVED** [review]
After a quick look without testing, looks completely fine.

File: src/editor/object_option.cpp (hunk 1 above)
Comment (MatusGuy) [reply src/editor/object_option.cpp:527]: What's with the square brackets?

Comment (mrkubax10) [reply src/editor/object_option.cpp:527]: Lambda function capture.

Comment (MatusGuy) [reply src/editor/object_option.cpp:527]: Ohhh those are aliases. Ok

**Pull request merged**